language/move-vm/runtime/src/session.rs

```diff
@@ -19,23 +20,26 @@ pub struct Session<'r, 'l, R> {
 }
 
 impl<'r, 'l, R: RemoteCache> Session<'r, 'l, R> {
-    pub fn execute_function(
+    pub fn execute_function<F: FnOnce(VMStatus) -> VMStatus>(
         &mut self,
         module: &ModuleId,
         function_name: &IdentStr,
         ty_args: Vec<TypeTag>,
         args: Vec<Value>,
         _sender: AccountAddress,
         cost_strategy: &mut CostStrategy,
-    ) -> VMResult<()> {
-        self.runtime.execute_function(
-            module,
-            function_name,
-            ty_args,
-            args,
-            &mut self.data_cache,
-            cost_strategy,
-        )
+        error_specializer: F,
```

Comment:
Correct, it just passes it through. 

I switched to this model because the conversions were ALL over the place. Some did it right after the function, some did it more upstream. 

This was the only way I could keep it sane

Reply:
yes I have no issue with this as is. There is stuff to review in terms of integration with the Move VM so this does not break or make things. And it's a good centralization of error handling